### PR TITLE
More high level APIs

### PR DESCRIPTION
**Pull request by IllusionPerdu**
I tried to continue the high level APIs.
The missing APIs were put in "todo", it may be possible to make only one macro with the macro "add_event_handle_host" and "add_event_handle" but I don't know how to do it ...
I think I cannot continue considering my little knowledge I have in rust.

### CODE REVIEW

**Comment by IllusionPerdu** [comment]
I would need help to finish.

File: src/lib.rs [hunk 1]
```diff
@@ -315,6 +414,101 @@ macro_rules! call {
     };
 }
 
+macro_rules! add_event_handle_host {
```
Comment (sopium) [inline src/lib.rs]: Can you rename this to `add_event_handler_host`(`handle` -> `handler`)? (And the same for the other two macros.)

**Comment by sopium** [comment]
Looks good. Anything in particular that you need help with?

File: src/lib.rs (hunk 1 above)
Comment (IllusionPerdu) [reply src/lib.rs]: ok it's done

**Comment by IllusionPerdu** [comment]
I discovered rust for 1 week.
For example, I'm not sure how to call:
* complete WebResourceRequestedEventArgs
* get_headers in WebResourceRequest and WebResourceResponse.
In fact practically all the todo that remains.

**Comment by sopium** [comment]
I am going to merge this and do a few more myself.

**Comment by IllusionPerdu** [comment]
thank you

**Pull request merged**